src/redis-cli.c

```diff
@@ -939,26 +1037,427 @@ static void completionCallback(const char *buf, linenoiseCompletions *lc) {
     }
 }
 
-/* Linenoise hints callback. */
-static char *hintsCallback(const char *buf, int *color, int *bold) {
-    if (!pref.hints) return NULL;
+static sds addHintForArgument(sds hint, cliCommandArg *arg);
```

Comment:
This hint helper code is so large that it could be considered to place it in a separate .c file. `redis-cli` is close to 10k lines.

Reply:
I agree, in principle. In practice, when I tried separating out the hint code it turned out to be more complicated than I expected, due to some interdependencies (I don't remember specifically which).

Reply:
OK, the globals are messy in this file. Leave it then. :+1: